File: guajira-bag/API/AuthController.cs
using GuajiraBags.Api.Models.DTOs;
using GuajiraBags.Api.Services;
using Microsoft.AspNetCore.Mvc;

namespace GuajiraBags.Api.Controllers;

[ApiController]
[Route("api/auth")]
public class AuthController(AuthService auth) : ControllerBase
{
    /// <summary>Registrar un nuevo usuario admin</summary>
    [HttpPost("register")]
    public async Task<IActionResult> Register([FromBody] RegisterRequest req)
    {
        if (!ModelState.IsValid) return BadRequest(ModelState);

        var (ok, error, response) = await auth.RegisterAsync(req);
        if (!ok) return BadRequest(new { error });

        return Ok(response);
    }

    /// <summary>Iniciar sesión y obtener JWT</summary>
    [HttpPost("login")]
    public async Task<IActionResult> Login([FromBody] LoginRequest req)
    {
        if (!ModelState.IsValid) return BadRequest(ModelState);

        var (ok, error, response) = await auth.LoginAsync(req);
        if (!ok) return Unauthorized(new { error });

        return Ok(response);
    }
}

File: Nueva carpeta/php-api/src/Controllers/AuthController.php
<?php

namespace GuajiraBags\Controllers;

use GuajiraBags\Helpers\Response;
use GuajiraBags\Services\AuthService;

class AuthController
{
    public function __construct(private AuthService $auth) {}

    public function register(): void
    {
        $body = json_decode(file_get_contents('php://input'), true) ?? [];
        [$ok, $error, $response] = $this->auth->register($body);

        if (!$ok) {
            Response::error($error, 400);
        }
        Response::json($response);
    }

    public function login(): void
    {
        $body = json_decode(file_get_contents('php://input'), true) ?? [];
        [$ok, $error, $response] = $this->auth->login($body);

        if (!$ok) {
            Response::error($error, 401);
        }
        Response::json($response);
    }
}
